src/mongocrypt-marking.c

```diff
@@ -564,6 +566,276 @@ _mongocrypt_fle2_placeholder_to_insert_update_ciphertext (
    return res;
 }
 
+// get_edges creates and returns edges from an FLE2RangeInsertSpec. Returns NULL
+// on error.
+static mc_edges_t *
+get_edges (mc_FLE2RangeInsertSpec_t *insertSpec,
+           size_t sparsity,
+           mongocrypt_status_t *status)
+{
+   bson_type_t value_type = bson_iter_type (&insertSpec->v);
+
+#define GET_EDGES(BITS)                                                       \
```

Comment:
I would rename to `GET_AND_RETURN_EDGES()` (since it returns) (and possible wrap with `if (1) { ... } else ((void)0)`)